### PR TITLE
Extended examples in API docs for QualName

### PR DESCRIPTION
**Pull request by Ygg01**
This PR adds documentation for `QualName` as well as some doc tests for it relating to #367 . @kornelski is this what you had in mind?

This PR is a minor change (since no API was changed other than the docs).

Comments and suggestions welcome.

### CODE REVIEW

File: markup5ever/interface/mod.rs [hunk 1]
```diff
@@ -119,21 +120,171 @@ pub mod tree_builder;
 ///    |    |
 ///    |    +- local name
 ///    |
-///  prefix (when resolved gives namespace_url)
+///  prefix (when resolved gives namespace_url `https://furniture.rs`)
 /// ```
+/// 
+/// NOTE: `Prefix`, `LocalName` and `Prefix` are all derivative of
```
Comment (SimonSapin) [inline markup5ever/interface/mod.rs]: This says `Prefix` twice. I assume one of them is intended to be `Namespace`.

**Comment by jdm** [comment]
@bors-servo r+
Nice improvement!

File: markup5ever/interface/mod.rs (hunk 1 above)
Comment (SimonSapin) [reply markup5ever/interface/mod.rs]: The rest of the diff looks good to me, feel free to merged after fixing this.

**Pull request merged**